src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs

```diff
@@ -139,8 +139,14 @@ public void DiscoverTests(DiscoveryCriteria discoveryCriteria, ITestDiscoveryEve
 
         _baseTestDiscoveryEventsHandler = eventHandler;
 
-        // Mark all sources as NotDiscovered before actual discovery starts
-        _discoveryDataAggregator.MarkSourcesWithStatus(discoverySources, DiscoveryStatus.NotDiscovered);
+        // Do not mark sources as not discovered, this is already done in the
```

Comment:
Note that marking source as not discovered would change nothing so maybe I could remove my comment.